shimmingtoolbox/dicom_to_nifti.py

```diff
@@ -0,0 +1,134 @@
+import os
+import numpy as np
+from .read_nii import read_nii
+
+
+def dicom_to_nifti(unsortedDicomDir, niftiPath):
+    """
+    TODO: indicate the purpose of this function
+    :param unsortedDicomDir:
+    :param niftiPath:
+    :return:
+    """
+
+    #os.mkdir(niftiPath)
+    print(unsortedDicomDir)
+    print(niftiPath)
+
+    # Check for which "which" to use depending on OS
+    if os.name == 'nt':  # If OS is Windows
+        which = 'where'
+        copy = 'copy'
+    else:
+        which = 'which'
+        copy = 'cp -r'
```

Comment:
Python has you covered for these sorts of common cross-platform issues. I mentioned `shutil.copytree` below.

As for `which`.. the only reason I suggested using that in matlab is that matlab doesn't have good error catching, especially not around non-matlab subprocesses, so I felt we needed to head off a common mistake. But in python I wouldn't bother with this at all; instead, just try to run it (with `subprocess.run(['dcm2niix', ...], check=True)`) and if it's not installed that call with crash and tell you.

Reply:
Before using os commands, you should always check if there is the command already implemented in a Pythonic way (most of the time, it is)

Reply:
I tried `shutil.copytree` but it wouldn't work because the path where it tries to copy the files already exists. I however found
`copy_tree` (from `distutils.dir_util`) that works just fine. Do you see any drawback in using that function @kousu  @gab-berest ?
More info about it [on that forum](https://stackoverflow.com/questions/15034151/copy-directory-contents-into-a-directory-with-python).

Reply:
I don't see a problem using this alternative. The problem I see is that you are trying to writesomething in a path that already exist. It could overwrite files that you don't want to overwrite. Normally you'd create a new folder in the path for your source data. This way, you are sure you don't delete something important (that's why shutil.copytree as this protection). At the end, it is also a lot easier to delete these data since they are in one folder.

Reply:
I the way the code is implemented, the folder where the data are copied (sourcedata) is created by dcm2bids right before the files are copied in, so there shouldn't be anything in there. I guess it's just a way for dcm2bids to make sure the folder names and organization are always the same.

Reply:
I see. then there should be no problem.